### PR TITLE
ref(processor): Try to fallback to previous frames token function name

### PR DESCRIPTION
**Pull request by kamilogorek**
This change applies the same heuristic that we previously used in the original `JavaScriptStacktraceProcessor`.
The rest is described in `get_function_for_token` function comment.

### CODE REVIEW

File: src/sentry/lang/javascript/processor.py [hunk 1]
```diff
@@ -861,19 +861,35 @@ def is_valid_frame(frame):
...
-def get_function_for_token(frame, token):
+def get_function_for_token(frame, token, previous_frame=None):
     """
     Get function name for a given frame, based on the looked up token.
-    Return tokens name if we have a usable value from symbolic or we have no initial function name at all,
-    otherwise, fallback to frames current function name.
+    Return tokens name if we have a usable value from symbolic,
+    if possible, ask the callsite whether it's token has a valid name if we
+    did not resolve to anything useful, otherwise, fallback to useless name
+    if we had no function name at all, or to frames current, minified function name.
```
Comment (loewenheim) [inline src/sentry/lang/javascript/processor.py:870]: This is a bit confusingly worded

**Review by loewenheim — APPROVED** [review]

File: src/sentry/lang/javascript/processor.py (hunk 1 above)
Comment (kamilogorek) [reply src/sentry/lang/javascript/processor.py:870]: Totally agree, I had a hard time coming up with something more reasonable 😅

Comment (loewenheim) [reply src/sentry/lang/javascript/processor.py:870]: Maybe a list of things that will be tried in order could work?

Comment (kamilogorek) [reply src/sentry/lang/javascript/processor.py:870]: Updated

**Review by loewenheim — APPROVED** [review]

File: src/sentry/lang/javascript/processor.py [hunk 2]
```diff
@@ -861,19 +861,36 @@ def is_valid_frame(frame):
...
-def get_function_for_token(frame, token):
+def get_function_for_token(frame, token, previous_frame=None):
     """
-    Get function name for a given frame, based on the looked up token.
-    Return tokens name if we have a usable value from symbolic or we have no initial function name at all,
-    otherwise, fallback to frames current function name.
+    Get function name for a given frame based on the token resolved by symbolic.
+    It tries following paths in order:
+    - return token function name if we have a usable value (filtered through `USELESS_FN_NAMES` list),
+    - return caller (previous frame) token function name if it had one mapped,
```
Comment (Swatinem) [inline src/sentry/lang/javascript/processor.py:869]: This should be:

> The SourceMapped `name` of the caller token.

```js
function caller(callee) {
  return callee();
  //           ^ stack frame points exactly here
}
caller(() => { throw new Error(); });
```

The stack trace points to the call expression `callee()` which should have a `name` mapping. That is what we want. We don’t want the function name (aka scope name) of the caller.

**Review by Swatinem — COMMENTED** [review]
I don’t think the logic here is correct

**Review by Swatinem — APPROVED** [review]

**Pull request merged**